src/main/scala/org/ergoplatform/db/KVStore.scala

```diff
@@ -0,0 +1,50 @@
+package org.ergoplatform.db
+
+import org.iq80.leveldb.{DB, ReadOptions}
+
+import scala.collection.mutable
+
+trait KVStore extends AutoCloseable {
+
+  type K = Array[Byte]
+  type V = Array[Byte]
+
+  protected val db: DB
+
+  def get(key: K): Option[V] =
+    Option(db.get(key))
+
+  def getAll(cond: (K, V) => Boolean): Seq[(K, V)] = {
```

Comment:
why not to convert iterator into seq in getAll()? The more flexible option is to return Iterator and then  it is up to client whether to read it fully or partially.

Reply:
Iterators are not used in the project, so all we got in case of using iterators is convertion overhead.